sharedbuffers/mapped_struct.py

```diff
@@ -344,13 +344,13 @@ def unpack_from(cls, buf, offs, idmap = None):
 
 
 @cython.ccall
-@cython.locals(code=cython.ulong, nbits=cython.ulong)
+@cython.locals(code=cython.ulonglong, nbits=cython.ulonglong)
 def _hash_rotl(code, nbits):
     return (code << nbits) | (code >> (64 - nbits))
 
 
 @cython.ccall
-@cython.locals(code1=cython.ulong, code2=cython.ulong)
+@cython.locals(code1=cython.ulonglong, code2=cython.ulonglong)
```

Comment:
Should also `@cython.returns(cython.ulonglong)` right?

Reply:
I thought cython was smart enough to figure that one out ...

Oh well, adding it won't hurt.

Reply:
It can probably infer it, but it will never change the function's signature and calling convention, which is what we're after.

Reply:
Also, unless you're unitt-testing these, you can use `cfunc` instead of `ccall` which is even lighter-weight. Since these are really tiny functions it probably helps.

You can also `cython.inline`.